neqo-transport/src/ecn.rs

```diff
@@ -9,18 +9,29 @@ use std::ops::{AddAssign, Deref, DerefMut, Sub};
 use enum_map::EnumMap;
 use neqo_common::{qdebug, qinfo, qwarn, IpTosEcn};
 
-use crate::{packet::PacketNumber, recovery::SentPacket};
+use crate::{
+    packet::{PacketNumber, PacketType},
+    recovery::SentPacket,
+};
 
 /// The number of packets to use for testing a path for ECN capability.
 pub const ECN_TEST_COUNT: usize = 10;
 
+/// The number of packets to use for testing the initial a path for ECN capability when exchanging
+/// Initials during the handshake. This is a lower number than `ECN_TEST_COUNT` to avoid
+/// unnecessarily delaying the handshake; we would otherwise double the PTO `ECN_TEST_COUNT` times.
+const ECN_TEST_COUNT_INITIAL_PHASE: usize = 3;
+
 /// The state information related to testing a path for ECN capability.
 /// See RFC9000, Appendix A.4.
 #[derive(Debug, PartialEq, Clone)]
 enum EcnValidationState {
     /// The path is currently being tested for ECN capability, with the number of probes sent so
     /// far on the path during the ECN validation.
-    Testing(usize),
+    Testing {
+        probes_sent: usize,
+        probes_lost: usize,
```

Comment:
```suggestion
        initial_probes_lost: usize,
```

These are not the probes lost in general, but the `PacketType::Initial` only, correct?